cmd/skaffold/app/cmd/debug.go

```diff
@@ -37,14 +46,70 @@ func NewCmdDebug() *cobra.Command {
 		WithLongDescription("Similar to `dev`, but configures the pipeline for debugging.").
 		WithCommonFlags().
 		WithHouseKeepingMessages().
+		WithFlags(func(f *pflag.FlagSet) {
+			// --filter and --build-artifacts are used to support debug for Helm and are internal
+			f.BoolVar(&filtering, "filter", false, `Filter manifests from stdin for debugging similar to "skaffold debug".`)
```

Comment:
nit: since `skaffold debug --filter` only writes the modified manifests to `StdOut` and doesn't do anything else should the flag be called `filter-only` or `render-only` or something which makes it obvious?

Reply:
I considered implementing this instead as a `skaffold filter --debug` command.  Maybe that's worth doing here, and we could add `--digests` as an option too.  WDYT @nkubala?